### DISCUSSION TITLE
keymap to select next matching word

### DISCUSSION THREAD

**Discussion by Bruce-Hopkins** [helix-editor/helix#9606] [General]
In VSCode, we have the keymap `ctrl+d`. It will find and select the next word that matches the currently selected word.

It is possible to do this within the current editor's features. You would have to select the entire text block, use `ctrl+s`, and search for the word you want to select. But this is not as clean or simple. 

I am willing to implement the feature myself.

**Comment by the-mikedavis** [comment]
For languages with a language server running, you can use `<space>h` to get a selection on all instances of the symbol under the cursor using LSP documentHighlight.

Otherwise, you can select a word, use `*` to save that selection to the search register, then use `vnnnn` to add selections for the next instances or use `%s<ret>` to select them all.

**Comment by Bruce-Hopkins** [comment]
> For languages with a language server running, you can use `<space>h` to get a selection on all instances of the symbol under the cursor using LSP documentHighlight.
> 
Yeah, that works. But I can't seem to find it in the documentation. Does that need to be added?

>Otherwise, you can select a word, use * to save that selection to the search register, then use vnnnn to add selections for the next instances or use %s<ret> to select them all.

I can't seem to get this to work. But maybe I'm doing it wrong.

**Comment by the-mikedavis** [comment]
Ah yep this is missing from the space mode docs and should be added https://docs.helix-editor.com/master/keymap.html#space-mode

You can try this in the languages.toml file in this repo: select "language" somewhere in the file, hit `*` to set the search register ('/') to the selection contents ("language"). Then go into select mode with `v` and hit `n` to add a new selection for the next match of "language" and `n` to add another. Or use `%` to select the full buffer and then `s` to subselect (this uses the search register by default). Hit Enter to perform the subselect and then you should have a selection for all "language" matches in the file

**Comment by erf** [comment]
I was about to suggest `<C-n>` as used in the [vis editor](https://github.com/martanne/vis) to create new selections / cursors from the current word, but i guess this is suggestion is similar and there are other ways to archive this.

I would still love this feature as i'm used to it and there are also many other cursor manipulation commands described in the [docs](http://martanne.github.io/vis/man/vis.1.html#Multiple_Selections) which might be of inspiration.

On another topic, i wish there was a way to create selections / cursors from a regex search without having to select all of the file content first using `%`

**Comment by xnuk** [comment]
Is there any updates or workarounds for this? I love to use `<c-d>`(select next matching word) and `<c-k>`(unselect current one and move to next one) in vscode/sublime, especially like when LSP server is not ready for renaming symbols yet.

**Comment by kenaryn** [comment]
Perhaps a sensitive-case flag would be useful when invoking search pattern when typing `s` key stroke.

**Comment by Rudxain** [comment]
- Related: #15491
